docs: exposing single server grafana with -p (#6235)
sourcegraph/sourcegraph: doc/admin/monitoring_and_tracing.md

### Single-container server deployments

If you are running Sourcegraph as single server you can add an additional publish flag for the Grafana port 3370:

> ```shell script

Reviewer: They still need `sshuttle` for accessing this on remote machines

Author: oh yeah

Author: done

> docker run --publish 7080:7080 --publish 2633:2633 --publish 127.0.0.1:3370:3370 \

Reviewer: You can do this in a different PR, but we could do this everywhere we suggest a `docker run` command like on https://docs.sourcegraph.com/ -- any reason you don't want to do that?

Reviewer: (then this section would just describe what to visit in your browser, and how to use `sshuttle` if running Sourcegraph on a remote server)

Author: sure, i'll scrub and add it to all

Author: done

### Single-container server deployments

> If you are running Sourcegraph as single server locally:

Reviewer: ```suggestion
If you are running Sourcegraph as a single-container server deployment:
```

```shell script
docker run --publish 7080:7080 --publish 2633:2633 --publish 127.0.0.1:3370:3370 \
  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \

>   --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3

Reviewer: Why are we still providing the docker run command here, and not the sshuttle command?

Author: seemed like a good place for a reminder how to run it :-). i'll remove it in a bit. am in the middle of some graphql metrics stuff on another branch

```

For simplicity, Garafana does not require authentication, as the port binding of 3370 is limited to connections from localhost.

> Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, a tool 

Reviewer 2: ```suggestion
Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, then access via an SSH tunnel using a tool
```

> like [sshuttle](https://github.com/sshuttle/sshuttle) to establish a secure connection to that URL. 

Reviewer 2: ```suggestion
such as [sshuttle](https://github.com/sshuttle/sshuttle) is required to establish a secure connection to Grafana.
```

> On your local machine start a `sshuttle` session to your remote server	

Reviewer 2: ```suggestion
To access the remote server using `sshuttle` from your local machine
```

> ```shell script	

Reviewer 2: ```suggestion
```sh
```

> sshuttle -r xxx_user_xxx@sxxx_remote_server_xxx 0/0

Reviewer 2: ```suggestion
sshuttle -r user@host 0/0
```

You can also use `ssh` to establish an SSH tunnel (although `sshuttle` is much more flexible and reliable):

```sh
SSH TUNNEL COMMAND GOES HERE
```

Reviewer 2: It's always good if possible, to provide a command that does not require the user to install a new tool. Obviously, if that's not the case for any reason, this doesn't apply.

```

> 

Reviewer: ```suggestion
Then simply visit ... in your browser.
```

### Single-container server deployments

> For simplicity, Garafana does not require authentication, as the port binding of 3370 is limited to connections from localhost.

Reviewer: ```suggestion
For simplicity, Garafana does not require authentication, as the port binding of 3370 is restricted to connections from localhost only.
```

Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, then access via an SSH tunnel using a tool
such as [sshuttle](https://github.com/sshuttle/sshuttle) is required to establish a secure connection to Grafana.
To access the remote server using `sshuttle` from your local machine:

```sh
sshuttle -r user@host 0/0
```

> Then simply visit ... in your browser.

Reviewer: ```suggestion
Then simply visit http://host:3370 in your browser.
```

I meant for you to replace ... with whatever is relevant, is this the correct URL or would it need /-/debug/ etc?

Reviewer: I see it redirects automatically, so the URL I suggested here is fine.

### Single-container server deployments

If you are running Sourcegraph as single server locally:

```shell script
docker run --publish 7080:7080 --publish 2633:2633 --publish 127.0.0.1:3370:3370 \
  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
```

> then http://localhost:3370/-/debug/grafana will just work. If you are running it on a remote server you can use a tool

Reviewer 2: ```suggestion
For simplicity, Garafana does not require authentication, as the port binding of 3370 is limited to connections from localhost.

Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana . If Sourcegraph is deployed to a remote server, a tool 
```
